src/main/java/org/broadinstitute/hellbender/tools/IndexFeatureFile.java

```diff
@@ -84,6 +84,78 @@ protected Object doWork() {
         return indexFile.getAbsolutePath();
     }
 
+    /**
+     * We want to report progress when indexing but the indexing code is in htsjdk and not available to us directly.
+     * The workaround is to make a special decorator 'codec' that gets called on every decoded Feature and this can use used to track progress.
+     * The codec delegates all calls and reports progress as it goes.
+     */
+    private static final class ProgressReportingDelegatingCodec<A extends Feature, B> implements FeatureCodec<A, B>{
```

Comment:
Don't embed this class within the indexing utility -- make public and move to `utils/codecs`